pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -396,6 +399,23 @@ func updateTaskRunsStatus(pr *v1alpha1.PipelineRun, pipelineState []*resources.R
 	}
 }
 
+func (c *Reconciler) updateTaskRunsStatusDirectly(pr *v1alpha1.PipelineRun) error {
```

Comment:
One note for the future (I'm pretty late to the party so not something we need to change now!) is that we generally try to avoid adding functionality at the reconciler level if we can: it's much easier to write (and read) test cases that don't require the entire reconciler to be setup (and the code tends to be simpler too)